numpy/core/multiarray.py

```diff
@@ -141,9 +141,9 @@ def empty_like(prototype, dtype=None, order=None, subok=None, shape=None):
 
 
 @array_function_from_c_func_and_dispatcher(_multiarray_umath.concatenate)
-def concatenate(arrays, axis=None, out=None):
+def concatenate(arrays, axis=None, out=None, dtype=None, casting=None):
```

Comment:
```suggestion
def concatenate(arrays, axis=None, out=None, *, dtype=None, casting=None):
```
to match my proposed C api change.

Note that we'll need #16298 sooner rather than later once we start adding keyword-only arguments.